src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -653,6 +653,23 @@ void
 PeerImp::doProtocolStart()
 {
     onReadMessage(error_code(), 0);
+
+    protocol::TMManifests tm;
+    tm.set_history (true);
+
+    overlay_.manifestCache ().for_each_manifest (
```

Comment:
Red flag here: `for_each_manifest` does it work while holding a lock. What's more, we're likely to need this very same message a whole lot of times, which means we'll rebuild it a whole lot of times. Does it make sense to generate it once, cache it and only rebuilding it when necessary?